tests/test_tags.py

```diff
@@ -0,0 +1,31 @@
+import asyncio
+
+import pytest
+import ucp
+
+
+@pytest.mark.asyncio
+async def test_tag_match():
+    msg1 = bytes("msg1", "utf-8")
+    msg2 = bytes("msg2", "utf-8")
+
+    async def server_node(ep):
+        f1 = ep.send(msg1, tag="msg1")
+        await asyncio.sleep(1)  # Let msg1 finish
+        f2 = ep.send(msg2, tag="msg2")
+        await asyncio.gather(f1, f2)
```

Comment:
Why do we have to `await asyncio.sleep(1)` above and then gather the same message here? Wouldn't the gather alone suffice?

Reply:
Looking below at the code, it seems like this is to enforce ordering, is that right?

Reply:
Yes, enforcing an order with calling `await` :)